neqo-transport/src/connection.rs

```diff
@@ -1178,17 +1182,21 @@ impl Connection {
         let mut needs_padding = false;
 
         // Check whether we are sending packets in PTO mode.
-        let (pto, cong_avail, min_pn_space) =
+        let (pto, cong_avail, min_pn_space, only_not_cc) =
             if let Some((min_pto_pn_space, can_send)) = self.loss_recovery.check_pto() {
                 if !can_send {
                     return Ok(None);
                 }
-                (true, path.mtu(), min_pto_pn_space)
+                (true, path.mtu(), min_pto_pn_space, false)
             } else {
+                // If avail == 0 we do not have available congestion window, we may send only
+                // non-congestion controlled frames
+                let avail = usize::try_from(self.loss_recovery.cwnd_avail()).unwrap();
                 (
                     false,
-                    usize::try_from(self.loss_recovery.cwnd_avail()).unwrap(),
+                    if avail == 0 { path.mtu() } else { avail },
                     PNSpace::Initial,
+                    avail == 0,
```

Comment:
totally a style thing but you might see if a structure like `if avail == 0 { // return a tuple} else { // return a different tuple }` might be clearer than one tuple whose elements are conditional on avail.